### PR TITLE
Update builder documentation to be platform-agnostic.

### PR DESCRIPTION
**Pull request by dgp1130**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [N/A] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [X] Documentation content changes

## What is the current behavior?

Builder CLI documentation references Unix commands such as `touch` and `ls`, which do not apply to Windows users.

Issue Number: #35373

## What is the new behavior?

Example is changed to simply copy a file rather than run arbitrary commands. This way it requires no knowledge of or support for Unix commands on the reader's machine.

## Does this PR introduce a breaking change?

- [X] No

### CODE REVIEW

File: aio/content/examples/cli-builder/src/my-builder.ts [hunk 1]
```diff
@@ -3,44 +3,42 @@
...
+): Promise<BuilderOutput> {
+  // #enddocregion builder, builder-skeleton, handling-output
+  // #docregion report-status
+  context.reportStatus(`Copying ${options.source} to ${options.destination}.`);
+  // #docregion builder, handling-output
+  try {
+    await fs.copyFile(options.source, options.destination);
+  } catch (err) {
+    // #enddocregion builder
+    context.logger.error(`Failed to copy file.`);
```
Comment (gkalpak) [inline aio/content/examples/cli-builder/src/my-builder.ts:28]:
```suggestion
    context.logger.error('Failed to copy file.');
```

Also, is the error message surfaced somewhere? Would it be helpful to include it here?

File: aio/content/examples/cli-builder/src/my-builder.ts [hunk 2]
```diff
@@ -3,44 +3,42 @@
...
-    });
-    // #docregion builder-skeleton
+    return {
+      success: false,
+      error: err.message,
+    };
+  }
+
+  // #enddocregion builder
+  context.reportStatus(`Done.`);
```
Comment (gkalpak) [inline aio/content/examples/cli-builder/src/my-builder.ts:37]:
```suggestion
  context.reportStatus('Done.');
```

Also, this line doesn't seem to belong to the `handling-output` docregion.

**Review by gkalpak — CHANGES_REQUESTED** [review]
Awesome re-work :tada:
(Just a few typos and minor suggestions.)

OOC, why `target: major`?

File: aio/content/examples/cli-builder/src/my-builder.spec.ts [hunk 3]
```diff
@@ -21,17 +22,12 @@ describe('Command Runner Builder', () => {
...
 
-  it('can run node', async () => {
-    // Create a logger that keeps an array of all messages that were logged.
-    const logger = new logging.Logger('');
-    const logs = [];
-    logger.subscribe(ev => logs.push(ev.message));
-
+  it('can copy files', async () => {
     // A "run" can have multiple outputs, and contains progress information.
     const run = await architect.scheduleBuilder('@example/command-runner:command', {
```
[the remark this answers is not in the document]
Comment (dgp1130) [reply aio/content/examples/cli-builder/src/my-builder.spec.ts]: Ah yes, I knew I would miss one of those!

File: aio/content/examples/cli-builder/src/my-builder.spec.ts [hunk 4]
```diff
@@ -41,8 +37,12 @@ describe('Command Runner Builder', () => {
     // to be scheduled.
     await run.stop();
 
-    // Expect that foo was logged
-    expect(logs).toContain('foo');
+    // Expect that the copied file is the same as its source.
+    const sourceContent = await fs.readFile('package.json', { encoding: 'utf8' });
+    const destinationContent = await fs.readFile('package-copy.json', {
+      encoding: 'utf8',
+    });
```
[the remark this answers is not in the document]
Comment (dgp1130) [reply aio/content/examples/cli-builder/src/my-builder.spec.ts]: Ah cool, I didn't know that!

File: aio/content/guide/cli-builder.md [hunk 5]
```diff
@@ -22,13 +22,13 @@ The separation of concerns here is the same as with [schematics](guide/glossary#
 
 * The `options` object is provided by the CLI user, while the `context` object is provided by the CLI Builder API.
 
-* In addition to the contextual information, the `context` object, which is an instance of the `BuilderContext`, also provides access to a scheduling method, `BuilderContext.scheduleTarget()`. The scheduler executes the builder handler function with a given [target configuration](guide/glossary#target).
+* In addition to the contextual information, the `context` object, which is an instance of the `BuilderContext`, also provides access to a scheduling method, `BuilderContext.prototype.scheduleTarget()`. The scheduler executes the builder handler function with a given [target configuration](guide/glossary#target).
```
[the remark this answers is not in the document]
Comment (dgp1130) [reply aio/content/guide/cli-builder.md]: Good point, replaced `BuilderContext.prototype` with `context`.

File: aio/content/examples/cli-builder/src/my-builder.ts (hunk 2 above)
Comment (dgp1130) [reply aio/content/examples/cli-builder/src/my-builder.ts]: Good catch.

File: aio/content/examples/cli-builder/src/my-builder.ts (hunk 1 above)
Comment (dgp1130) [reply aio/content/examples/cli-builder/src/my-builder.ts]: The error gets returned and architect shows it to the user, this is just providing some additional context for the following error message.

**Comment by dgp1130** [comment]
> OOC, why `target: major`?

I was trying to say that since this is a docs change it only needs to merge into `master`. Is there a different label I should use here?

**Comment by clydin** [comment]
master is `target: minor` currently.  That will change to `target: major` when master is in v13 prerelease.

**Comment by dgp1130** [comment]
Actually, I forgot this was on aio rather than a simple README. Would this make more sense as a patch release? I'm not sure where exactly we draw the line between "bug" and "feature" for docs?

**Comment by clydin** [comment]
Looks like pretty much all the docs PRs are for `target: patch` so that probably makes sense here as well

File: aio/content/guide/cli-builder.md [hunk 6]
```diff
@@ -333,16 +335,16 @@ If we create a new project with `ng new builder-test`, the generated `angular.js
...
-This target will tell the builder to run `touch` on a file, in order to update its modified date.
+Let's add a new target that will run our builder to copy a file.
+This target will tell the builder to copy the `package.json` file.
 
 We need to update the `angular.json` file to add a target for this builder to the "architect" section of our new project.
 
 * We'll add a new target section to the "architect" object for our project.
 
-* The target named "touch" uses our builder, which we published to `@example/command-runner`. (See [Publishing your Library](guide/creating-libraries#publishing-your-library).)
+* The target named "copy" uses our builder, which we published to `@example/copy-file`. (See [Publishing your Library](guide/creating-libraries#publishing-your-library))
```
Comment (clydin) [inline aio/content/guide/cli-builder.md:345]:
```suggestion
* The target named "copy-package" uses our builder, which we published to `@example/copy-file`. (See [Publishing your Library](guide/creating-libraries#publishing-your-library))
```

File: aio/content/guide/cli-builder.md [hunk 7]
```diff
@@ -333,16 +335,16 @@ If we create a new project with `ng new builder-test`, the generated `angular.js
...
 
-* The options object provides default values for the two inputs that we defined; `command`, which is the Unix command to execute, and `args`, an array that contains the file to operate on.
+* The options object provides default values for the two inputs that we defined; `source`, which is the existing file we are copying, and `destination`, the path we want to copy to.
 
 * The configurations key is optional, we'll leave it out for now.
```
Comment (clydin) [inline aio/content/guide/cli-builder.md:349]:
```suggestion
* The `configurations` key is optional, we'll leave it out for now.
```

**Review by clydin — APPROVED** [review]
Nice job.
The copy file concept is a good choice to reduce the complexity of the example.

Just two comments.

**Review by gkalpak — APPROVED** [review]
🚀

**Pull request closed without merging**